Update password is done

asliddintursunov opened the pull request:
Commit №42: Porfile update is done, updated some backend codes and need to update frontend too so that there will not be confusion during api fetches

asliddintursunov committed:
Update password is done

asliddintursunov committed:
Made better code, almost nothing added to UI

asliddintursunov commented:
Commit №43: Almost nothing has been added to UI, however changed some functions/function logics. Removed not used functions and added new error handlings.

muslimchikkk commented on app/auth/login/page.tsx:
```diff
@@ -1,14 +1,13 @@
 /* eslint-disable react-hooks/exhaustive-deps */
```
It is always good not to disable eslint rules. They are enabled for some reason

muslimchikkk commented on app/auth/login/page.tsx:
```diff
@@ -22,25 +21,40 @@ export default function Login() {
   const [password, setPassword] = useState<string>("");
 
   useEffect(() => {
-    const handleLogin = async () => {
```
I would refactor this hook to make it more readable and clean. Have a look at SOLID principles to learn how to approach code quality: https://www.digitalocean.com/community/conceptual-articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design

muslimchikkk commented on app/auth/login/page.tsx:
```diff
@@ -22,25 +21,40 @@ export default function Login() {
-          const res = await axios.post(`${baseURL}/auth/login`, {
-            username,
-            password,
+          const request = await fetch(`${baseURL}/auth/login`, {
```
The API fetch functions should always be in some other file rather than a react component. You could create axios instance with a config that you want and reuse it.

asliddintursunov replied:
Yeah, initially all fetch functions were on the file and the reason that I moved them to the react component is, because they were used only 1 times. 2 or more times used functions are on the other file. If writing fetch functions on the other file even better, I'll do it, no problem :)

muslimchikkk commented on app/auth/register/form/page.tsx:
```diff
@@ -26,29 +26,44 @@ export default function Register() {
-            username,
-            email,
-            password,
+          const request = await fetch(`${baseURL}/auth/register`, {
```
My comments above applies here as well

muslimchikkk commented on app/auth/register/interests/components/Interests.tsx:
```diff
@@ -34,16 +34,32 @@ function Interests({}: Props) {
-        username: newUserUsername,
-        interests: selectedInterests,
+      const request = await fetch(`${baseURL}/auth/set_interests`, {
+        method: "POST",
```
Let's reuse some pre-configured axios instance everywhere

asliddintursunov committed:
Made code better look, Deleted not used functions on lib directory

asliddintursunov committed:
Fixed removing extra characters on title

muslimchikkk commented on app/[username]/(pages)/home/components/HomePagePosts.tsx:
```diff
@@ -101,9 +106,9 @@ function HomePagePosts({ username, firstTenUserPosts }: Props) {
-                    `/tag/${el.post_primary_category}/${el.post_title}_${el.post_uuid}`
-                      .replaceAll(" ", "-")
-                      .toLowerCase()
+                    `/tag/${el.post_primary_category}/${formatTitleForUrl(
```
nice!

asliddintursunov committed:
Editor is updated to medium like editor. Used tiptap editor

asliddintursunov committed:
created fetchFunction, Auth files and put all api fetch functions there so that I can use them later, Updated editor component and added some new features.

asliddintursunov commented:
On commit №47. I could not add pre-configured axios instance everywhere because most of my components are server component and any other HTTP requests libraries cannot be used on server-side compoenent. 

**New updated features are**

- updated editor, and now it looks like a medium editor
- moved fetch functions into one file and used them where is needed
